src/primitives/Avatar.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useMemo } from "react";
+import DefaultLogo from "@/assets/default_logo.png";
+import { match, P } from "ts-pattern";
+import { Avatar as ShadCNAvatar, AvatarFallback, AvatarImage } from "@/ui-shadcn/avatar";
+
+interface AvatarProps {
+  fallbackName?: string;
+  ipfsCID?: string;
+  url?: string;
+  size?: number;
+  ipfsBaseURL?: string;
+  defaultImage?: string;
+}
+
+export const Avatar = ({
+  fallbackName,
+  ipfsCID,
+  url,
+  size = 40,
+  ipfsBaseURL = "https://ipfs.io/ipfs/",
+  defaultImage = DefaultLogo,
+}: AvatarProps) => {
+  const imageURL = useMemo(() => {
+    return match({ ipfsCID, url, fallbackName })
+      .with({ ipfsCID: P.nullish, url: P.nullish, fallbackName: P.nullish }, () => defaultImage)
+      .with({ ipfsCID, url: P.string.length(0) }, ({ ipfsCID }) => `${ipfsBaseURL}${ipfsCID}`)
+      .with({ ipfsCID, url: P.nullish }, ({ ipfsCID }) => `${ipfsBaseURL}${ipfsCID}`)
+      .with({ ipfsCID: P.string.length(0), url }, ({ url }) => url)
+      .with({ ipfsCID: P.nullish, url }, ({ url }) => url)
+      .with(
+        { ipfsCID: P.string.minLength(1), url: P.string.minLength(1) },
+        ({ ipfsCID }) => `${ipfsBaseURL}${ipfsCID}`,
+      )
+
+      .otherwise(() => defaultImage);
+  }, [ipfsCID, url, ipfsBaseURL]);
+
+  const fallback = useMemo(() => {
+    return match(fallbackName)
+      .with(undefined, () => "")
+      .otherwise((name) => {
+        const words = name.trim().split(/\s+/);
+        return match(words)
+          .with([P.string], ([word]) =>
+            word.length === 1 ? word.toUpperCase() : word.slice(0, 2).toUpperCase(),
+          )
+          .otherwise(([first, second]) => (first[0] + (second?.[0] || "")).toUpperCase());
+      });
+  }, [fallbackName]);
+
+  return (
+    <ShadCNAvatar
+      role="presentation"
+      className="ui-h-full ui-w-full ui-shadow-md ui-shadow-slate-600"
+      style={{ width: `${size}px`, height: `${size}px` }}
```

Comment:
let's stick to tailwind. cleaning up later will be harder

Reply:
Because shadcn define hardcoded width and height in tailwind as 40 😬 So if we define it here, it's ignored.  

Ill remove the size-40 from shadcn so it's just tailwind, that means if we ever update the shadcn component the size test will fail and we'll have to remove it again

Reply:
I don't understand this part @gnomadic  let's talk about it later so I can understand the issue you refer from shadcn

Reply:
here: https://github.com/gitcoinco/gitcoin-ui/blob/main/src/ui-shadcn/avatar.tsx#L13

it's `ui-h-10 and `ui-w-10` (not 40, my bad). but basically if I pass in a different height or width through tailwind it gets ignored the one there is used.  I can remove those from shadcn no big deal, that's the point of shadcn doing the "copy" method of installation

Reply:
<img width="917" alt="Screenshot 2024-10-21 at 9 09 53 AM" src="https://github.com/user-attachments/assets/5e32947c-747a-4aa3-a7b0-3524a0538e60">

dang it doesn't work, looks like issue with JIT with tailwind.  I think if we only use this when we have to it's ok